test/test_unistd.rs

```diff
@@ -576,6 +577,60 @@ fn test_symlinkat() {
     );
 }
 
+
+#[test]
+#[should_panic]
+fn test_unlinkat_dir_noremovedir() {
+    let tempdir = tempfile::tempdir().unwrap();
+    let dirname = "foo_dir";
+    let dirpath = tempdir.path().join(dirname);
+
+    // Create dir
+    DirBuilder::new().recursive(true).create(&dirpath).unwrap();
+
+    // Get file descriptor for base directory
+    let dirfd = fcntl::open(tempdir.path(), fcntl::OFlag::empty(), stat::Mode::empty()).unwrap();
+
+    // Attempt unlink dir at relative path without proper flag
+    unlinkat(Some(dirfd), dirname, UnlinkatFlags::NoRemoveDir).unwrap();
+    assert!(dirpath.exists());
```

Comment:
This line is dead code, assuming the line above panics.

Reply:
Thanks. You are right. Not sure why I left this in but it is now removed.